subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/transform/ImmutableCachingTransformationWorkspaceProvider.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.internal.artifacts.transform;
+
+public class ImmutableCachingTransformationWorkspaceProvider extends DefaultCachingTransformationWorkspaceProvider {
+    public ImmutableCachingTransformationWorkspaceProvider(TransformationWorkspaceProvider delegate) {
+        super(delegate);
+    }
+}
```

Comment:
I got sucked into a black hole with services while thinking about this, and I have no good solution for this. But it still feels wrong to have the same exact type under two different names intentionally.